superbench/runner/runner.py

```diff
@@ -460,14 +462,19 @@ def run(self):
                             continue
                         with open(self._output_path / 'hostfile', 'r') as f:
                             host_list = f.read().splitlines()
-                        pattern_hostx = gen_traffic_pattern_host_group(host_list, mode.pattern)
-                        for host_groups in pattern_hostx:
-                            para_rc_list = Parallel(n_jobs=len(host_groups))(
-                                delayed(self._run_proc)
-                                (benchmark_name, mode, vars={
-                                    'proc_rank': 0,
-                                    'host_list': host_group,
-                                }) for host_group in host_groups
+                        host_groups = gen_traffic_pattern_host_groups(host_list, mode.pattern)
+                        for serial_index, host_group in enumerate(host_groups):
```

Comment:
I forgot to input the path parameter here so it looks weird. pattern config file will be generated under output folder for each run, and it will be a mandatory operation if user specified pattern in mpi mode, so that they can use that file to diagnose. For the second question, I think the config will not change in each run for all benchmarks as hostfile is not really changed. Duplication config (same pattern for different benchmarks) will be considered in the future.

Reply:
Agree with you, as it will be better for helping users to further diagnose. Will update.